src/box/box.cc

```diff
@@ -3635,9 +3693,13 @@ local_recovery(const struct tt_uuid *instance_uuid,
 		box_update_replication();
 
 		struct replica *master;
-		if (replicaset_needs_rejoin(&master)) {
+		while (replicaset_needs_rejoin(&master)) {
 			say_crit("replica is too old, initiating rebootstrap");
-			return bootstrap_from_master(master);
+			if (bootstrap_from_master_begin(master) == 0) {
+				bootstrap_from_master_end(master);
+				return;
+			}
+			box_restart_replication();
```

Comment:
1 - Is this retrying tested?
2 - I suggest to add a `fiber_sleep()` here, otherwise you are potentially making a busy loop;
3 - Need a log that you couldn't rejoin first time and will retry every N secs;

Reply:
All done